dockers/docker-database/create_all_redis_config.sh.j2

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+{% if DATABASE %}
+{% for redis_inst in DATABASE %}
```

Comment:
Can we use a straightforward runtime loop instead of rending a template?

Reply:
Runtime loop will be easier. ref: https://stackoverflow.com/a/1955555/2514803

Reply:
> Runtime loop will be easier by a python oneliner. ref: https://stackoverflow.com/a/1955555/2514803

two questions: 
1. no other shell script use this way to pasre json file so far in SONiC. Even we use this python method to get the context, we still need parse it to find out useful information in shell. We may need to call python again to make the codes easy.

2. Then why we don't use python directly instead of shell if anyway we need to use python?

3. I didn't see rending a  template cause any issues, the codes is also easy to understand and reuse the existing sonic-cfggen?

Reply:
I see your point. In your case, you are trying to generate several redis.conf according to a /etc/sonic/database_config file.

The easiest way may be a single python script. You are using a bash template to generate a bash script, then run another bash script to manipulate the default redis.conf. It's too complex for this intent.

---
In reply to: [241636378](https://github.com/Azure/sonic-buildimage/pull/2182#discussion_r241636378) [](ancestors = 241636378)